// @ualterjr/ujrclock/dist/esm/es2017/build/ujr-clock.sc.entry.js
/*! Built with http://stenciljs.com */
import { h } from '../ujrclock.core.js';

class UjrClock {
    constructor() {
        this.time = Date.now();
    }
    componentDidLoad() {
        this.timer = window.setInterval(() => {
            this.time = Date.now();
        }, 1000);
        console.log(this.format);
    }
    componentDidUnload() {
        window.clearInterval(this.timer);
    }
    render() {
        var now = new Date();
        var time = now.toLocaleTimeString('en-US', { hour12: true });
        if (this.format == "24") {
            time = now.toLocaleTimeString('en-US', { hour12: false });
        }
        return (h("span", null,
            time,
            " \u00A0\u00A0\u00A9 Ujr"));
    }
    static get is() { return "ujr-clock"; }
    static get properties() { return {
        "format": {
            "type": String,
            "attr": "format"
        },
        "time": {
            "state": true
        }
    }; }
    static get style() { return "span{font-family:Arial Narrow}"; }
}

export { UjrClock };
